sqlmesh/dbt/loader.py

```diff
@@ -77,11 +100,110 @@ def _load_models(
         for package in project.packages.values():
             context.variables = package.variables
             package_models: t.Dict[str, BaseModelConfig] = {**package.models, **package.seeds}
+
             models.update(
-                {model.model_name: model.to_sqlmesh(context) for model in package_models.values()}
+                {
+                    model.model_name: cache.get_or_load_model(
+                        model.path, lambda: model.to_sqlmesh(context)
+                    )
+                    for model in package_models.values()
+                }
             )
 
         return models
 
     def _load_audits(self) -> UniqueKeyDict[str, Audit]:
         return UniqueKeyDict("audits")
+
+    def _compute_yaml_max_mtime_per_subfolder(self, root: Path) -> t.Dict[Path, float]:
+        if not root.is_dir():
+            return {}
+
+        result = {}
+        max_mtime: t.Optional[float] = None
+
+        for nested in root.iterdir():
+            if nested.is_dir():
+                result.update(self._compute_yaml_max_mtime_per_subfolder(nested))
+            elif nested.suffix.lower() in (".yaml", ".yml"):
+                yaml_mtime = self._path_mtimes.get(nested)
+                if yaml_mtime:
+                    max_mtime = max(max_mtime, yaml_mtime) if max_mtime is not None else yaml_mtime
+
+        if max_mtime is not None:
+            result[root] = max_mtime
+
+        return result
+
+    class _Cache:
```

Comment:
damn dbt has it's only cache ?

Reply:
sorry, not sure I understand the comment

Reply:
just complaining that it seems like we needed two similar classes

Reply:
yeah, there's potentially a way to consolidate them but doesn't seem like it's worth the investment right now.